Source/WebCore/Modules/modern-media-controls/controls/button.css

```diff
@@ -55,10 +55,16 @@ button:focus {
 }
 
 button:focus > picture {
-    background-color: -webkit-focus-ring-color !important;
     mix-blend-mode: normal !important;
 }
 
+/* In GTK/WPE clicking a button brings focus to it, so in that case we want to avoid setting the
+   background color to focus-ring-color. That's why this CSS rule applies only to non-adwaita
+   layouts. */
+button:is(.mac, .ios, .watchos):focus > picture {
```

Comment:
Ah, this isn't quite what I was suggesting in my previous comment 😅

I meant more something like this
```
:is(.mac, .ios, .watchos) button:focus > picture {
```

Reply:
That's what I tried first indeed, but the layout tests were still failing. After 2 hours of head-scratching this is what I came up with.

Reply:
Which tests were the ones that were failing?  If I had to guess, it's possibly because the tests that directly use `Button` weren't part of a parent that had the necessary CSS class (i.e. it's expecting something like `<div class="mac">...<button></button>...</div>`), but maybe there was some other issue.  I can say for certain though that the `.mac`/`.ios`/`.watchos` CSS class is added by some `*MediaControls` (i.e. it's not directly on the `<button>`).

Reply:
LayoutTests/media/modern-media-controls/button/button-focus-state.html was so I had make the button CSS class explicit there, and also this change in the button.css. Your suggestion didn't work out of the box, I have no idea why, but this special incantation I came up with seems to have made the EWS happy!

Anyways, I suppose this is a hack in the end. That's why I marked this PR as draft. Thanks anyways for taking the time to review the patch!

Reply:
yeah that test creates a `Button` directly

i'd probably do something like this
```diff
-document.body.appendChild(button.element);
+
+// Simulate being a child of `MacOSInlineMediaControls` as that's needed for some styles.
+let wrapper = document.body.appendChild(document.createElement("div"));
+wrapper.className = "mac";
+wrapper.appendChild(button.element);
```